### ISSUE TITLE
[NoMethodError] undefined method on intellisense

### ISSUE THREAD

**Issue by ws1088** [castwide/solargraph#463]
```
[Error - 1:48:00 PM] Request textDocument/completion failed.
  Message: [NoMethodError] undefined method `start' for nil:NilClass
  Code: -32603 
```
Hi got this error when doing simple ruby coding like when I press `.` after `t1`
```
t1 = 'abcd'
t1.
```
I am using the latest gems. What could be the issue? Thanks in advance!

**Comment by castwide** [comment]
Is there a backtrace in the developer tools console (Help -> Toggle Developer Tools)?

**Comment by ws1088** [comment]
Here you go. Thanks!

```
console.ts:137 [Extension Host] [WARN] Error processing request: [NoMethodError] undefined method `start' for nil:NilClass
console.ts:137 [Extension Host] [WARN] 
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:131:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/source.rb:218:in `string_ranges'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/source.rb:185:in `string_at?'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/source/cursor.rb:103:in `string?'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/source_map/clip.rb:26:in `complete'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/library.rb:166:in `completions_at'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/host.rb:513:in `completions_at'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/message/text_document/completion.rb:12:in `process'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/host.rb:103:in `receive'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/adapter.rb:44:in `process'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/adapter.rb:17:in `block in opening'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/data_reader.rb:59:in `parse_message_from_buffer'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/data_reader.rb:35:in `block in receive'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/data_reader.rb:30:in `each_char'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/data_reader.rb:30:in `receive'
[...]/.gem/ruby/2.6.0/gems/solargraph-0.42.4/lib/solargraph/language_server/transport/adapter.rb:27:in `receiving'
[...]/.gem/ruby/2.6.0/gems/backport-1.2.0/lib/backport/client.rb:66:in `tick'
[...]/.gem/ruby/2.6.0/gems/backport-1.2.0/lib/backport/server/tcpip.rb:82:in `update'[...]/pkgs/ruby/2.6.0/lib/ruby/2.6.0/observer.rb:197:in `block in notify_observers'[...]//pkgs/ruby/2.6.0/lib/ruby/2.6.0/observer.rb:196:in `each'[...]/pkgs/ruby/2.6.0/lib/ruby/2.6.0/observer.rb:196:in `notify_observers'
[...]/.gem/ruby/2.6.0/gems/backport-1.2.0/lib/backport/client.rb:124:in `read_input'
[...]/.gem/ruby/2.6.0/gems/backport-1.2.0/lib/backport/client.rb:105:in `block in run_input_thread'
```

**Comment by castwide** [comment]
I haven't been able to reproduce this error, but my best guess is that it's a problem parsing an interpolated string, e.g., `"#{var}"` or `$( #{var} )`. If the problem persists in version 0.44.0, can you possibly give me a simple example that causes it?

**Comment by win93** [comment]
Another test case where this occurs, if this is the entire file's contents:

```ruby
%W[array of words #{'with a substitution'}]
```

Then typing anywhere in the file triggers the same error with the following stack trace:

```
[Extension Host] [WARN] Error processing request: [NoMethodError] undefined method `start' for nil:NilClass
[WARN] /Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:131:in `string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/source.rb:218:in `string_ranges'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/source.rb:185:in `string_at?'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/source/cursor.rb:103:in `string?'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/source_map/clip.rb:26:in `complete'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/library.rb:166:in `completions_at'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/language_server/host.rb:515:in `completions_at'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/language_server/message/text_document/completion.rb:14:in `process'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/language_server/host.rb:112:in `receive'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/language_server/host/message_worker.rb:53:in `tick'
/Users/alexg/.gem/ruby/3.0.0/gems/solargraph-0.44.0/lib/solargraph/language_server/host/message_worker.rb:44:in `block in start'
```

**Comment by gukandrew** [comment]
same here, osx big sur, ruby 2.6.0 over asdf, solargraph version 0.44.0 used in vscode last version
```
[INFO] Server received textDocument/completion
[WARN] Error processing request: [NoMethodError] undefined method `start' for nil:NilClass
[WARN] /Users/andrew/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:131:in `string_ranges'
/Users/andrew/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
/Users/andrew/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `each'
/Users/andrew/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:133:in `string_ranges'
/Users/andrew/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/solargraph-0.44.0/lib/solargraph/parser/rubyvm/class_methods.rb:134:in `block in string_ranges'
```

**Comment by pramoditagi** [comment]
@castwide any updates on this?

**Comment by opan** [comment]
@castwide Any update on this issue? I have started to get this issue couple of days ago. Previously work just fine. I don't remember recent changes in my local machine that caused the issue.

Using WSL2, Ubuntu 20.04. Ruby 2.6.0 (Rbenv) 
```❯ 💎 v2.6.0 gem info solargraph

*** LOCAL GEMS ***

solargraph (0.44.2)
    Author: Fred Snyder
    Homepage: http://solargraph.org
    License: MIT
    Installed at: /home/opan/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0

    A Ruby language server
```

**Comment by castwide** [comment]
Sorry for the late delay. There's a fix for this committed to master. I expect to release it in a new version in the next day or so.

**Comment by castwide** [comment]
Released in v0.45.0.
